src/main/scala/akka/contrib/process/NonBlockingProcess.scala

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.contrib.process
+
+import akka.actor.{ Actor, ActorLogging, NoSerializationVerificationNeeded, PoisonPill, Props, Status }
+import akka.stream.scaladsl.{ BroadcastHub, Keep, Sink, Source }
+import akka.util.ByteString
+import java.io.File
+import java.nio.ByteBuffer
+
+import akka.stream.{ ActorMaterializer, OverflowStrategy }
+import akka.{ Done, NotUsed }
+import com.zaxxer.nuprocess.{ NuAbstractProcessHandler, NuProcess, NuProcessBuilder }
+
+import scala.collection.JavaConverters
+import scala.collection.immutable
+import scala.concurrent.{ Future, blocking }
+
+object NonBlockingProcess {
+
+  /**
+   * Sent to the receiver on startup - specifies the streams used for managing input, output and error respectively.
+   * This message should only be received by the parent of the NonBlockingProcess and should not be passed across the
+   * JVM boundary (the publishers are not serializable).
+   *
+   * @param pid the process id
+   * @param stdin a `akka.stream.scaladsl.Sink[ByteString]` for the standard input stream of the process
+   * @param stdout a `akka.stream.scaladsl.Source[ByteString]` for the standard output stream of the process
+   * @param stderr a `akka.stream.scaladsl.Source[ByteString]` for the standard error stream of the process
+   */
+  case class Started(
+    pid: Long,
```

Comment:
Yeah, I was influenced here by the forthcoming JDK process API (which we'll probably never use!)